shared/adapters/chain/cosmos/keys.ts

```diff
@@ -1,31 +1,20 @@
 // Cosmos cannot sign arbitrary blobs, but they can sign transactions. So, as a hack around that,
 // we insert our account registration token into a proposal message, and then verify against the
 // generated signature. But first we need the message to insert.
```

Comment:
See https://github.com/cosmos/cosmjs/issues/844 and the linked cosmos-sdk resources for more insights around this issue. Happy to collaborate on some good enough hack spec shared by clients.

Reply:
Makes sense. I believe it was my suggestion from several years ago which raised the issue to begin with: https://github.com/cosmos/cosmos-sdk/issues/4581 -- I still think signing a submit proposal tx should work as a hack, since it does support arbitrary text data in the title/description period, but please let me know if you have other ideas.